catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/forms/builder/JsonModelBuilder.java

```diff
@@ -207,23 +206,32 @@ public JsonModelBuilder endTerminalType() {
   public JsonModelBuilder setProperty(String property) {
     verifyResultNotYetRetrieved();
     verifyTerminalNodeInProgress();
-    nodeInProgress.setProperty(property);
+
+    nodeInProgress.accept(new PropertySettingVisitor(property));
+
     return this;
   }
 
+  @Override
+  public FlatFilterBuilder setLiteralProperty(Object literalProperty) {
+    throw new RuntimeException("not implemented");
```

Comment:
the method `setLiteralProperty` is only called when going from json to xml. Seems smelly that we have a method that isn't actually used. What do you think @Lambeaux ?